rero_ils/modules/api.py

```diff
@@ -39,13 +39,32 @@
 from invenio_records_rest.utils import obj_or_import_string
 from invenio_search import current_search
 from invenio_search.api import RecordsSearch
+from jsonschema import FormatChecker
+from jsonschema.compat import str_types
 from jsonschema.exceptions import ValidationError
 from kombu.compat import Consumer
 from sqlalchemy import text
 from sqlalchemy.orm.exc import NoResultFound
 
+from .constants import DEFAULT_EMAIL_REGEXP
 from .utils import extracted_data_from_ref
 
+"""Custom ILS record JSON schema format validator."""
+ils_record_format_checker = FormatChecker()
+
+
+@ils_record_format_checker.checks('email')
+@ils_record_format_checker.checks('idn-email')
+def _strong_email_validation(instance) -> bool:
+    """Allow to validate an email address (only email format, not DNS)."""
+    if not isinstance(instance, str_types):
+        return False
+    email_regexp = DEFAULT_EMAIL_REGEXP
+    if current_app:  # could be use outside of the application context
```

Comment:
Would it not be easier to import `EMAIL_VALIDATION_REGEXP` directly from the config.py file?

Reply:
If we put the DEFAULT_EMAIL_REGEXP into the `config.py` the app context should be required. Isn't it ?

Reply:
> If we put the DEFAULT_EMAIL_REGEXP into the `config.py` the app context should be required. Isn't it ?

Yes but you can have two different behaviour depending if you have the app context or not...

Reply:
After concertation, I choose to place the email validation regexp into the function.
The reason is that it seems not relevant (not highly needed) to have a special configuration to validate email using a config setting. Placing this setting into the config forces to include `app` fixture to validate json schema using an 'email' field.
This solution is still much better than the current `jsonschema` library basic behavior : validate anything except if is a string. In this case, the validation only requires an "@" character.
```python
@_checks_drafts(name="idn-email")
@_checks_drafts(name="email")
def is_email(instance):
    if not isinstance(instance, str_types):
        return True
    return "@" in instance
 ```